src/api/CortexClient.ts

```diff
@@ -381,18 +386,27 @@ export class CortexClient implements CortexApi {
     input: RequestInfo,
     init?: RequestInit,
   ): Promise<Response> {
-    let token: string | undefined = undefined;
+    let token: string | undefined;
+    let email: string | undefined;
+    let displayName: string | undefined;
     if (this.identityApi !== undefined) {
       ({ token } = await this.identityApi.getCredentials());
+      const profileInfo = await this.identityApi.getProfileInfo();
+      email = profileInfo.email;
+      displayName = profileInfo.displayName;
     }
 
+    const headers = {
+      ...init?.headers,
+      Authorization: `Bearer ${token}`,
+      'x-cortex-email': email ? email : '',
+      'x-cortex-name': displayName ? displayName : '',
```

Comment:
nit: can be simplified using [nullish coalescing](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing)
```suggestion
      'x-cortex-email': email ?? '',
      'x-cortex-name': displayName ?? '',
```